### PR TITLE
(#4071) - Improve code coverage 

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by daleharvey** [comment]
I split this into its own PR since its a little less sure whether we want to go this way, so the vast majority of these are handling error conditions that we never get while running the tests and shouldnt be expecting to get at all, instrumenting the errors to exercise the code paths is going to be hard to somewhat impossible and if we remove the callbacks then it makes life harder for when something does go disastrously wrong

**Comment by daleharvey** [comment]
Ok this is like a pick and mix, they are all seperate commits so if there are issues with some the rest can still land, all together this gets us to `Lines        : 99.37% ( 4420/4448 )`

**Comment by nolanlawson** [comment]
Awesome work, really happy to see the coverage go up, whether it's through dead code removal or reasonable `istanbul ignore`s.

File: lib/adapters/leveldb/index.js [hunk 1]
```diff
@@ -145,6 +147,7 @@ function LevelPouch(opts, callback) {
   }
 
   if (typeof leveldown.destroy !== 'function') {
+    /* istanbul ignore next */
     leveldown.destroy = function (name, cb) { cb(); };
   }
```
Comment (nolanlawson) [inline lib/adapters/leveldb/index.js]: ^ FYI this was an old fix for old MemDOWN and probably isn't required anymore.

File: lib/adapter.js [hunk 2]
```diff
@@ -279,6 +281,7 @@ AbstractPouchDB.prototype.removeAttachment =
       callback(errors.error(errors.REV_CONFLICT));
       return;
     }
+    /* istanbul ignore if */
     if (!obj._attachments) {
       return callback();
     }
```
Comment (nolanlawson) [inline lib/adapter.js]: ^ this is weird and should probably just be removed.

**Pull request merged**